src/lib/prisma-session-store.ts

```diff
@@ -221,6 +242,70 @@ export class PrismaSessionStore<M extends string = 'session'> extends Store {
         `Attempt to destroy non-existent session:${String(sid)} ${String(e)}`
       );
       if (callback) defer(callback, e);
+
+      return;
+    }
+
+    if (callback) defer(callback);
+  };
+
+  /**
+   * Destroy all session(s) for the user(s) associated with the provided `sid`(s)'.
+   * Prerequisite:
+   *   All sessions in the store have been created using set() with a
+   *  `session` argument containing the `uid` (user id / name) property.
+   *
+   * @param sid a single or multiple id(s) to remove data for
+   * @param callback a callback notifying that the session(s) have
+   * been destroyed or that an error occurred
+   */
+  public readonly destroyUsersSessions = async (
+    sid: string | string[],
+    callback?: (err?: unknown) => void
+  ) => {
+    if (!(await this.validateConnection())) return callback?.();
+
+    try {
+      // Find all unique user ids for the sessions specified by the sid argument
+      const sids = Array.isArray(sid) ? sid : [sid];
+      const uids: { [key: string]: boolean } = {};
+      for (const id of sids) {
+        const s = await this.prisma[this.sessionModelName].findUnique({
+          where: { sid: id },
+        });
+        if (s === null) {
+          const errMsg = `No session found for provided session id: ${String(
+            sid
+          )}`;
+          this.logger.warn(errMsg);
+          throw Error(errMsg);
+        }
+        const uid = s.uid;
+        if (typeof uid !== 'string') {
```

Comment:
Should we check for an empty string here also? Since uid isn't required/unique, could people set it to `@default('')`, which would then result in this clearing the sessions for _all_ users who don't have the uid explicitly set

Reply:
Good point.